airflow-core/src/airflow/cli/commands/connection_command.py

```diff
@@ -39,26 +39,46 @@
 from airflow.utils import cli as cli_utils, helpers, yaml
 from airflow.utils.cli import suppress_logs_and_warning
 from airflow.utils.db import create_default_connections as db_create_default_connections
+from airflow._shared.secrets_masker import redact
 from airflow.utils.providers_configuration_loader import providers_configuration_loaded
 from airflow.utils.session import create_session
 
 
-def _connection_mapper(conn: Connection) -> dict[str, Any]:
-    return {
+def _connection_mapper(conn: Connection, show_values: bool = True, hide_sensitive: bool = False) -> dict[str, Any]:
+    """
+    Map connection to dictionary for display.
+    
+    :param conn: Connection object
+    :param show_values: If False, hide sensitive fields (password, URI, extra)
+    :param hide_sensitive: If True, mask sensitive fields instead of showing them
+    """
+    result = {
         "id": conn.id,
         "conn_id": conn.conn_id,
         "conn_type": conn.conn_type,
         "description": conn.description,
         "host": conn.host,
         "schema": conn.schema,
         "login": conn.login,
-        "password": conn.password,
         "port": conn.port,
         "is_encrypted": conn.is_encrypted,
         "is_extra_encrypted": conn.is_encrypted,
-        "extra_dejson": conn.extra_dejson,
-        "get_uri": conn.get_uri(),
     }
+    
+    # Only include sensitive fields if show_values is True
+    if show_values:
+        if hide_sensitive:
+            # Use Airflow's secrets masker to redact sensitive values
+            result["password"] = redact(conn.password, name="password")
```

Comment:
Look at `redact` -> you should not specify name, when you have dict, redact will recursively pass name from keys of the dictionary and replace all sensitive values automatically. name-None is default. So it is way better if you redact the whole "result" dictionary - this will be way simpler:

1) prepare dictionary
2) if hide_sensitive -> redact the dictionary